flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/result/CollectResultBase.java

```diff
@@ -29,23 +29,28 @@
 
 /** A result that works through {@link TableResult#collect()}. */
 public abstract class CollectResultBase implements DynamicResult {
-    private final CloseableIterator<RowData> result;
+    private final CloseableIterator<RowData> resultIterator;
 
     protected final Object resultLock;
     protected AtomicReference<SqlExecutionException> executionException = new AtomicReference<>();
     protected final ResultRetrievalThread retrievalThread;
 
     public CollectResultBase(TableResultInternal tableResult) {
-        result = tableResult.collectInternal();
+        resultIterator = tableResult.collectInternal();
         resultLock = new Object();
         retrievalThread = new ResultRetrievalThread();
     }
 
     @Override
-    public void close() throws Exception {
+    public void close() {
```

Comment:
Why changes to catch the exception internally?

Reply:
Because all invokers ignore the errr, so I think it would much easier we ignore this inside here.